src/serac/physics/tests/serac_thermal_functional.cpp

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/thermal_conduction_functional.hpp"
+
+#include <fstream>
+
+#include <gtest/gtest.h>
+#include "mfem.hpp"
+
+#include "serac/mesh/mesh_utils.hpp"
+#include "serac/physics/state/state_manager.hpp"
+#include "serac/serac_config.hpp"
+
+namespace serac {
+
+template <int p, int dim>
+void functional_test_static(double expected_temp_norm)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  int serial_refinement   = 1;
+  int parallel_refinement = 0;
+
+  // Create DataStore
+  axom::sidre::DataStore datastore;
+  serac::StateManager::initialize(datastore, "serac", "thermal_functional_static_solve");
+
+  static_assert(dim == 2 || dim == 3, "Dimension must be 2 or 3 for thermal functional test");
+
+  // Construct the appropriate dimension mesh and give it to the data store
+  if constexpr (dim == 2) {
+    std::string meshfile2D = SERAC_REPO_DIR "/data/meshes/star.mesh";
+    auto mesh2D = mesh::refineAndDistribute(buildMeshFromFile(meshfile2D), serial_refinement, parallel_refinement);
+
+    serac::StateManager::setMesh(std::move(mesh2D));
+  }
+
+  if constexpr (dim == 3) {
+    std::string meshfile3D = SERAC_REPO_DIR "/data/meshes/beam-hex.mesh";
+    auto mesh3D = mesh::refineAndDistribute(buildMeshFromFile(meshfile3D), serial_refinement, parallel_refinement);
+
+    serac::StateManager::setMesh(std::move(mesh3D));
+  }
```

Comment:
I don't think `if constexpr` is necessary here, maybe something like 

```suggestion

  std::map<int, std::string> meshfiles = {
    {2, SERAC_REPO_DIR "/data/meshes/star.mesh"},
    {3, SERAC_REPO_DIR "/data/meshes/beam_hex.mesh"}  
  };
  
  auto mesh = mesh::refineAndDistribute(buildMeshFromFile(meshfiles[dim]), serial_refinement, parallel_refinement);
  serac::StateManager::setMesh(std::move(mesh));
```

or 

```cpp
std::string filename = (dim == 2) ? 
  SERAC_REPO_DIR "/data/meshes/star.mesh" : 
  SERAC_REPO_DIR "/data/meshes/beam-hex.mesh";
auto mesh = mesh::refineAndDistribute(buildMeshFromFile(filename), serial_refinement, parallel_refinement);
serac::StateManager::setMesh(std::move(mesh));

Reply:
I'll add the second option.